joomla/joomla-cms
# [6.2] Make abstract immutable event actually immutable

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/39372 .

### Summary of Changes
As pointed out in the issue our immutable event is very mutable. This fixes all the cases so that once initialised there can be no properties set into the event (it’s possible we might need some plug-in events to actually be mutable).

As we've now split the Immutable class from the mutable class - I've taken this opportunity to move the create method to it's own factory class - because it will be somewhat confusing for users that the mutable AbstractEvent class can create AbstractImmutable class's which are somewhat unrelated.

### Testing Instructions
Ensure core continues to work when performing a variety of actions. We also should test this with some 3rd party plugins to ensure they also still work.

### Actual result BEFORE applying this Pull Request
Event is mutable

### Expected result AFTER applying this Pull Request
Event is not mutable

### Link to documentations
Please select:
